### ISSUE TITLE
php52 classmap support

### ISSUE THREAD

**Issue by staabm** [composer/composer#1730]
ATM a autoload_classmap looks like

``` php
// autoload_classmap.php generated by Composer

$vendorDir = dirname(__DIR__);
$baseDir = dirname($vendorDir);

return array(
);
```

which is not php5.2 compilant (because the use of `__DIR__`).

I know that composer requires PHP5.3.2+, but in case composer would generate this only file in a way which works on PHP < 5.3.2 we could introduce composer also to our legacy production systems.

This would work when meeting the premises
- handle only dependencies which are also compilant with php < 5.3.2 
- use a custom autoloader which knows how to handle this classmap (because composers' autoloader requires 5.3+ due to the namespaces)
- use php 5.3+ on the boxes which invoke the composer command itself

TL;DR having a stripped down runtime of composer which is 5.\* compilant would also ease things in legacy systems.
